schema/ext/vulnerability-2.0-SNAPSHOT.schema.json

```diff
@@ -0,0 +1,235 @@
+      "enum": [
+        "None",
+        "Low",
+        "Medium",
+        "High",
+        "Critical",
+        "Unknown"
+      ]
+    },
+    "scoreMethod": {
+      "type": "string",
+      "title": "method",
+      "description": "Specifies the risk scoring methodology/standard used.",
+      "enum": [
+        "CVSSv2",
+        "CVSSv3",
+        "OWASP Risk",
+        "Open FAIR",
+        "Other"
+      ]
+    },
+    "score": {
+      "oneOf": [
+        {
+          "type": "number"
+        },
+        {
+          "type": "object",
+          "title": "Score",
+          "description": "Defines the numerical risk score of a vulnerability",
+          "properties": {
+            "base": {
+              "type": "number",
+              "title": "Base Score",
+              "description": "The base score of the security vulnerability (Refer CVSS standard for example)",
+              "multipleOf": 0.1,
+              "minimum": 0,
+              "maximum": 10,
+              "examples": [
+                2.9,
+                7.2
+              ]
+            },
+            "impact": {
+              "type": "number",
+              "title": "Impact Score",
+              "description": "The impact subscore of the security vulnerability (Refer CVSS standard for example)",
+              "multipleOf": 0.1,
+              "minimum": 0,
+              "maximum": 10,
+              "examples": [
+                2.9,
+                7.2
+              ]
+            },
+            "exploitability": {
+              "type": "number",
+              "title": "Exploitability Score",
+              "description": "The exploitability subscore of the security vulnerability (Refer CVSS standard for example)",
+              "multipleOf": 0.1,
+              "minimum": 0,
+              "maximum": 10,
+              "examples": [
+                2.9,
+                7.2
+              ]
+            }
+          }
+        }
+      ]
+    },
+    "rating": {
+      "type": "object",
+      "title": "Rating",
+      "description": "Defines the risk ratings of a vulnerability.",
+      "properties": {
+        "score": {
+          "$ref": "#/definitions/score"
+        },
+        "severity": {
+          "$ref": "#/definitions/severity"
+        },
+        "method": {
+          "$ref": "#/definitions/ratingMethod"
+        },
+        "source": {
+          "type": "string",
+          "title": "Source",
+          "description": "The entity responsible for providing the rating",
+          "examples": [
+            "NVD"
+          ]
+        },
+        "vector": {
+          "type": "string",
+          "title": "Vector",
+          "description": "Textual representation of the metric values used to score the vulnerability see attack vector in https://www.first.org/cvss/v3.1/specification-document"
```

Comment:
Are you not assuming then that all rating systems use CVSS and a vector?

Reply:
None of these are required fields, which is a bit of a design decision with CycloneDX I feel based on some of the questions I asked previously. So ratings here can be any of these individually or together.